### PR TITLE
Feature deleteuser

### PR DESCRIPTION
**Pull request by suryasinghvijay**
Please review.
need to test one flow (delete a logged-out user) will test it once logout API is available.

### CODE REVIEW

File: quora-api/src/main/java/com/upgrad/quora/api/controller/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @param authorization
+     * @param userId UUID of the user who has to be deleted from db
+     * @return UUID of the deleted user
+     * @throws AuthorizationFailedException unAuthorization exception is thrown when non admin user tries to delete the User
+     * @throws UserNotFoundException thrown when trying to delete un-registered user
+     */
+    @RequestMapping(method = RequestMethod.DELETE, path = "/admin/user/{userId}", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<UserDeleteResponse> deleteUser(@RequestHeader("authorization") final String authorization, @PathVariable String userId) throws AuthorizationFailedException, UserNotFoundException {
+        String deletedUserUUID = authenticationService.checkUserByAuthToken(authorization, userId);
```
Comment (arunsubashm) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/AdminController.java:30]: Can the service name be changed as deleteAuthUser() to make it more related to the function it does

File: quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java [hunk 2]
```diff
@@ -12,18 +14,44 @@
...
+     * @param request   WebRequest
+     * @return ResponseEntity
+     */
+    @ExceptionHandler(UserNotFoundException.class)
+    public ResponseEntity<ErrorResponse> userNotFoundException(UserNotFoundException exception, WebRequest request) {
+        return new ResponseEntity<ErrorResponse>(new ErrorResponse().code(exception.getCode()).message(exception.getErrorMessage()), HttpStatus.NOT_FOUND);
+    }
+
+    /**
+     * Handles AuthorizationFailedException. Triggered when condition not met during deleting the user.
```
Comment (arunsubashm) [inline quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java]: comment updated as - User not found

File: quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java [hunk 3]
```diff
@@ -80,5 +80,31 @@ public UserAuthTokenEntity createAuthToken(final UserAuthTokenEntity userAuthTok
...
         entityManager.merge(updateUserEntity);
     }
+
+    /**This method deletes the user and flush's it.
+     *
+     * @param userId UUID to be deleted
+     * @return true
+     */
+    public boolean deleteUser(String userId) {
+        entityManager.createQuery("DELETE FROM UserEntity c WHERE c.uuid = :uuid")
```
Comment (arunsubashm) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java]: Can this also be made as a named query

**Review by arunsubashm — CHANGES_REQUESTED** [review]

File: quora-api/src/main/java/com/upgrad/quora/api/controller/AdminController.java (hunk 1 above)
Comment (suryasinghvijay) [reply quora-api/src/main/java/com/upgrad/quora/api/controller/AdminController.java:30]: There are 2 table's that get affect by this
1. UserEntity (users)
2. User Auth table
so made it generic as delete user as it does not depend only on User auth table.

Do you want me to go head and update it please let me know

File: quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java (hunk 3 above)
Comment (arunsubashm) [reply quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java]: Thanks for implementing

**Review by arunsubashm — APPROVED** [review]

**Pull request merged**